Documentation for get/set CLI and DBus Config/Try/Cancel/Apply/Get/Set

slyon opened the pull request:
## Description
Add man pages for `netplan get/set` CLI and `Config/Try/Cancel/Apply/Get/Set` DBus methods.
This PR also adds a `dbus-config` feature flag, to indicate the availability of the DBus configuration management.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [x] New/changed keys in YAML format are documented.
- [x] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

sil2100 reviewed (changes requested):
The resulting manpages look good, but I'd like it if you could include information about the config invalidation - as per my inline comment.

Post 0.101 I think it would be also nice if we included some examples for people (probably somehow in `examples`? Though right now we only have `.yaml` configs there...), showing a typical DBus interaction scenario. Since then it would be easier for them to understand how the interactions with a config object are supposed to be performed to get a modified config applied from start to the end.

slyon commented on doc/netplan-dbus.md:
```diff
@@ -0,0 +1,39 @@
+The ``/io/netplan/Netplan/config/<ID>`` objects provide a ``io.netplan.Netplan.Config`` interface, offering the following methods:
+
+ * ``Get() -> s``: calls **netplan get --root-dir=/tmp/netplan-config-ID all** and returns the merged YAML config of the the given config object's state
+ * ``Set(s:CONFIG_DELTA, s:ORIGIN_HINT) -> b``: calls **netplan set --root-dir=/tmp/netplan-config-ID --origin-hint=<ORIGIN_HINT> CONFIG_DELTA**, where CONFIG_DELTA can be something like: ``network.ethernets.eth0.dhcp4=true`` and ORIGIN_HINT can be something like: ``70-snapd`` (it will then write the config to ``70-snapd.yaml``)
```
I've extended the description of `Set()` a bit, to explain the conflict  management.

slyon commented:
Thank you! I've updated the `netplan-dbus.8` manpage accordingly and also added a brief DBus config management scenario in `examples/dbus_config_scenario.txt`, by showing the control flow via `busctl` commands.

sil2100 reviewed (approved):
Sweet! Thank you!
